src/Business Foundation/App/NoSeries/src/Batch/NoSeriesBatch.Codeunit.al

```diff
@@ -0,0 +1,206 @@
+    begin
+        exit(NoSeriesBatchImpl.PeekNextNo(NoSeriesLine, UsageDate));
+    end;
+    #endregion
+    /// <summary>
+    /// Get the last number used in the No. Series.
+    /// Note: If a line was just closed, this function will return an empty string. Please use the NoSeriesLine overload to get the Last number for closed lines.
+    /// </summary>
+    /// <param name="NoSeriesCode">Code for the No. Series.</param>
+    /// <returns>The last number used in the series.</returns>
+    procedure GetLastNoUsed(NoSeriesCode: Code[20]): Code[20]
+    begin
+        exit(NoSeriesBatchImpl.GetLastNoUsed(NoSeriesCode));
+    end;
+
+    /// <summary>
+    /// Get the last number used in the No. Series.
+    /// </summary>
+    /// <param name="NoSeriesLine">The No. Series line to use.</param>
+    /// <returns>The last number used in the series.</returns>
+    procedure GetLastNoUsed(var NoSeriesLine: Record "No. Series Line"): Code[20]
+    begin
+        exit(NoSeriesBatchImpl.GetLastNoUsed(NoSeriesLine));
+    end;
+
+    /// <summary>
+    /// Verifies that the No. Series allows using manual numbers.
+    /// Note: This function allows manual numbers for blank No. Series Codes.
+    /// </summary>
+    /// <param name="NoSeriesCode">Code for the No. Series.</param>
+    procedure TestManual(NoSeriesCode: Code[20])
+    var
+        NoSeries: Codeunit "No. Series";
+    begin
+        NoSeries.TestManual(NoSeriesCode);
+    end;
+
+    /// <summary>
+    /// Verifies that the No. Series allows using manual numbers and throws an error for the document no. if it does not.
+    /// Note: This function allows manual numbers for blank No. Series Codes.
+    /// </summary>
+    /// <param name="NoSeriesCode">Code for the No. Series.</param>
+    /// <param name="DocumentNo">Document No. to be shown in the error message.</param>
+    procedure TestManual(NoSeriesCode: Code[20]; DocumentNo: Code[20])
+    var
+        NoSeries: Codeunit "No. Series";
+    begin
+        NoSeries.TestManual(NoSeriesCode, DocumentNo);
+    end;
+
+    /// <summary>
+    /// Simulate the specified No. Series at the specified date starting with the indicated number.
+    /// </summary>
+    /// <param name="NoSeriesCode">Code for the No. Series.</param>
+    /// <param name="UsageDate">The date of retrieval, this will influence which line is used.</param>
+    /// <param name="LastNoUsed">Simulate this is the last number used.</param>
+    /// <returns></returns>
+    procedure SimulateGetNextNo(NoSeriesCode: Code[20]; UsageDate: Date; LastNoUsed: Code[20]): Code[20]
+    var
+        NoSeriesBatchImplSim: Codeunit "No. Series - Batch Impl.";
+    begin
+        exit(NoSeriesBatchImplSim.SimulateGetNextNo(NoSeriesCode, UsageDate, LastNoUsed));
+    end;
+
+    /// <summary>
+    /// Puts the codeunit in simulation mode which disables the ability to save state.
+    /// </summary>
+    procedure SetSimulationMode()
```

Comment:
Do we want to have a boolean parameter here? E. g. if someone wants to first set the mode to simulation, and then change it back.